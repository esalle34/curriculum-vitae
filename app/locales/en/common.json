{
  "title": "Eric Salle",
  "submit": "Submit",
  "chooseLanguage": "Language & Country",
  "close": "Close"
}
