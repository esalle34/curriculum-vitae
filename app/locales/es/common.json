{
  "title": "Eric Salle",
  "submit": "Confirmar",
  "chooseLanguage": "Lengua i Pa√≠s",
  "close": "Cerrar"
}
