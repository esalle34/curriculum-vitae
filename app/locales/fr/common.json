{
  "title": "Eric Salle",
  "submit": "Valider",
  "chooseLanguage": "Langue & Pays",
  "close": "Fermer"
}
