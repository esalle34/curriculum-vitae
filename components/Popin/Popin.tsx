import { Dispatch, PropsWithChildren, SetStateAction } from "react";
import Image from "next/image";

interface Props {
  displayed: boolean;
  setDisplayed: Dispatch<SetStateAction<boolean>>;
  title: string;
  closeText: string;
}

const Popin = ({
  displayed,
  children,
  setDisplayed,
  title,
  closeText,
}: PropsWithChildren<Props>) => (
  <div
    data-testid="popin"
    className={
      displayed
        ? "absolute w-full h-full flex justify-center items-center"
        : "hidden"
    }
  >
    <div className="flex flex-col gap-2 md:w-1/4 bg-slate-100 py-4 px-6 rounded z-10 border-2 border-slate-200">
      <div className="flex justify-between items-center">
        <h3 className="py-2 text-lg font-semibold text-sky-800">{title}</h3>
        <Image
          onClick={() => setDisplayed(false)}
          className="cursor-pointer"
          src="/images/cross.svg"
          width={20}
          height={20}
          alt={closeText}
          title={closeText}
          role="img"
        />
      </div>
      {children}
    </div>
    <div
      data-testid="background"
      onClick={() => setDisplayed(false)}
      className={
        displayed ? "absolute w-full h-full bg-slate-400 opacity-50" : "hidden"
      }
    />
  </div>
);

export default Popin;
