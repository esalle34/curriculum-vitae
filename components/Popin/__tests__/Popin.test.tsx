import { fireEvent, render } from "@testing-library/react";
import Popin from "..";
import * as dict from "@/app/locales/fr/common.json";

describe("Popin", () => {
  it("should match snapshot correctly", () => {
    const { container } = render(
      <Popin
        displayed
        setDisplayed={jest.fn()}
        title={dict.chooseLanguage}
        closeText={dict.close}
      >
        <></>
      </Popin>
    );
    expect(container).toMatchSnapshot();
  });
  it("should match snapshot correctly with display false", () => {
    const { container } = render(
      <Popin
        displayed={false}
        setDisplayed={jest.fn()}
        title={dict.chooseLanguage}
        closeText={dict.close}
      >
        <></>
      </Popin>
    );
    expect(container).toMatchSnapshot();
  });
  it("should close Popin on click on closeSVG", () => {
    const setDisplayed = jest.fn();
    const { getByRole } = render(
      <Popin
        displayed={true}
        setDisplayed={setDisplayed}
        title={dict.chooseLanguage}
        closeText={dict.close}
      >
        <></>
      </Popin>
    );
    const closeSvg = getByRole("img");
    fireEvent.click(closeSvg);
    expect(setDisplayed).toHaveBeenCalled();
  });
  it("should close Popin on click outside", () => {
    const setDisplayed = jest.fn();
    const { getByTestId } = render(
      <Popin
        displayed={true}
        setDisplayed={setDisplayed}
        title={dict.chooseLanguage}
        closeText={dict.close}
      >
        <></>
      </Popin>
    );
    const background = getByTestId("background");
    fireEvent.click(background);
    expect(setDisplayed).toHaveBeenCalled();
  });
});
