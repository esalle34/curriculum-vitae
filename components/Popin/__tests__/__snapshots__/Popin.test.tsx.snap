// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popin should match snapshot correctly 1`] = `
<div>
  <div
    class="absolute w-full h-full flex justify-center items-center"
    data-testid="popin"
  >
    <div
      class="flex flex-col gap-2 md:w-1/4 bg-slate-100 py-4 px-6 rounded z-10 border-2 border-slate-200"
    >
      <div
        class="flex justify-between items-center"
      >
        <h3
          class="py-2 text-lg font-semibold text-sky-800"
        >
          Langue & Pays
        </h3>
        <img
          alt="Fermer"
          class="cursor-pointer"
          data-nimg="1"
          decoding="async"
          height="20"
          loading="lazy"
          role="img"
          src="/images/cross.svg"
          style="color: transparent;"
          title="Fermer"
          width="20"
        />
      </div>
    </div>
    <div
      class="absolute w-full h-full bg-slate-400 opacity-50"
      data-testid="background"
    />
  </div>
</div>
`;

exports[`Popin should match snapshot correctly with display false 1`] = `
<div>
  <div
    class="hidden"
    data-testid="popin"
  >
    <div
      class="flex flex-col gap-2 md:w-1/4 bg-slate-100 py-4 px-6 rounded z-10 border-2 border-slate-200"
    >
      <div
        class="flex justify-between items-center"
      >
        <h3
          class="py-2 text-lg font-semibold text-sky-800"
        >
          Langue & Pays
        </h3>
        <img
          alt="Fermer"
          class="cursor-pointer"
          data-nimg="1"
          decoding="async"
          height="20"
          loading="lazy"
          role="img"
          src="/images/cross.svg"
          style="color: transparent;"
          title="Fermer"
          width="20"
        />
      </div>
    </div>
    <div
      class="hidden"
      data-testid="background"
    />
  </div>
</div>
`;
